metronome/checkpointing/models/src/io/iohk/metronome/checkpointing/models/MerkleTree.scala

```diff
@@ -1,14 +1,30 @@
 package io.iohk.metronome.checkpointing.models
 
 import io.iohk.metronome.core.Tagger
+import io.iohk.metronome.crypto.hash.Keccak256
 import scodec.bits.ByteVector
 
+import scala.annotation.tailrec
+
+sealed trait MerkleTree {
+  def hash: MerkleTree.Hash
+}
+
 object MerkleTree {
   object Hash extends Tagger[ByteVector] {
-    val empty = apply(ByteVector.empty)
+
+    /** Indicates an empty MerkleTree */
+    val empty = apply(Keccak256(ByteVector.empty))
```

Comment:
A request for comments: this was initially just an empty `ByteVector` (not hashed), but e.g. Ethereum does choose a 32-byte value for the case of empty set. I guess the question boils down to do we need this value to be of constant size?

Reply:
The way for example Dmitry modeled the `Hash` type is that it asserts that it is of 32 bytes long, so I think an all zero hash, or this hash of an empty vector, would be better than an empty hash.